File: html/cgi/php/welcome.php
<?php
// Check if cookies are set
error_reporting(E_ALL);

if (isset($_COOKIE['bgcolor']) && isset($_COOKIE['user'])) {
    $name = htmlspecialchars($_COOKIE['user']);
    $bgcolor = htmlspecialchars($_COOKIE['bgcolor']);
    $title = $name;
    $message = "Welcome to your panel !";
} else {
    $name = "Unknown User";
    $bgcolor = "LightGoldenRodYellow";
    $title = "Unknown User";
    $message = "Oops! You're not connected yet";
}

// Build the HTML body
$body = "<!DOCTYPE html>
<html>
<head>
    <title>$title</title>
    <style>
        * { font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif; }
        body { background-color: $bgcolor; display: flex; justify-content: center; align-items: center; height: 100vh; margin: 0; flex-direction: column; }
        h1 { font-size: 3em; color: white; }
    </style>
</head>
<body>
    <h1>$name</h1>
    <h2>$message</h2>";

// Add buttons based on user connection status
if (isset($_COOKIE['bgcolor']) && isset($_COOKIE['user'])) {
    $body .= "<br><a href=\"/php/ColorModifier.html\"><button> Change Your Profile Color</button></a>
    <br><a href=\"/php/LogOut.php\"><button> Log Out </button></a>";
} else {
    $body .= "<br><a href=\"/php/connectionForm.html\"><button> Create a new profile</button></a>";
}

$body .= "</body></html>";

// Print the HTTP headers
echo "HTTP/1.1 200 OK\r\n";
echo "Content-Type: text/html\r\n";
echo "Content-Length: " . strlen($body);
echo "\r\n";

// Print the HTML body
echo "\r\n$body\r\n";

// Print server name and date
if (isset($_SERVER['SERVER_NAME'])) {
    $serverName = $_SERVER['SERVER_NAME'];
    echo "Server: $serverName \r\n";
}

$date = gmdate("D, d M Y H:i:s T");
echo "Date: $date\r\n";
?>

File: html/cgi/perl/welcome.pl
#!/usr/bin/perl
use strict;
use warnings;
use POSIX qw(strftime);

# Retrieve environment variables
my $cookie_string = $ENV{'HTTP_COOKIE'} || '';
my $server_name = $ENV{'SERVER_NAME'} || 'Unknown Server';

# Format date to match Python output
my $date = strftime "%a, %d %b %Y %T %Z", localtime;

# Initialize HTML body
my $body = "<!DOCTYPE html>";

# Parse cookies manually
my %cookies = map { split /=/, $_, 2 } split /;\s*/, $cookie_string;

if (exists $cookies{'bgcolor'} && exists $cookies{'user'}) {
    my $name = $cookies{'user'};
    my $bgcolor = $cookies{'bgcolor'};
    $body .= <<"END_HTML";
<head><title>$name</title>
<style>
* {font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}
body { background-color: $bgcolor; display: flex; justify-content: center; align-items: center; height: 100vh; margin: 0; flex-direction: column;}
h1 { font-size: 3em; color: white; }
</style></head>
<body>
<h1>$name</h1>
<h2>Welcome to your panel !</h2>
<br><a href="/perl/ColorModifier.html"><button> Change Your Profile Color</button></a>
<br><a href="/perl/perls.pl"><button> Go check the number of perls you have there! </button><a>
<br><a href="/perl/LogOut.pl"><button> Log Out </button></a>
</body></html>
END_HTML
} else {
    $body .= <<"END_HTML";
<head><title> Unknown User</title>
<style>
* {font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;}
body { background-color: LightGoldenRodYellow; display: flex; justify-content: center; align-items: center; height: 100vh; margin: 0; flex-direction: column;}
h1 { font-size: 3em; color: black; }
</style></head>
<body>
<h1>Oops ! You're not connected yet</h1>
<br><a href="/perl/connectionForm.html"><button> Create a new profile</button></a>
</body></html>
END_HTML
}

# Print server and date info
print "Server: $server_name\n";
print "Date: $date\n";

# Print the HTTP headers and body content
print "Content-Type: text/html\n";
print "Content-Length: " . length($body) . "\r\n\r\n";
print $body;
